packages/models-library/src/models_library/user_preferences.py

```diff
@@ -0,0 +1,177 @@
+            raise NoPreferenceFoundError(msg)
+        return preference_class
+
+    @classmethod
+    def get_preference_name(cls) -> PreferenceName:
+        # NOTE: this will be `unique` among all subclasses.
+        # No class inherited from this one, can be defined using the same name,
+        # even if the context is different.
+        return cls.__name__
+
+    @classmethod
+    def get_default_value(cls) -> Any:
+        return (
+            cls.__fields__["value"].default_factory()
+            if cls.__fields__["value"].default_factory
+            else cls.__fields__["value"].default
+        )
+
+    def dict(  # noqa: A003
+        self,
+        *,
+        include: AbstractSetIntStr | MappingIntStrAny = None,  # type: ignore
+        exclude: AbstractSetIntStr | MappingIntStrAny = None,  # type: ignore
+        by_alias: bool = False,
+        skip_defaults: bool = None,  # type: ignore  # noqa: RUF013
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+    ) -> DictStrAny:
+        if exclude is None:
+            config_class = getattr(self, "Config", None)
+            exclude_from_serialization: set | None = getattr(
+                config_class, "exclude_from_serialization", None
+            )
+            if exclude_from_serialization:
+                exclude = exclude_from_serialization
+        return super().dict(
+            include=include,
+            exclude=exclude,
+            by_alias=by_alias,
+            skip_defaults=skip_defaults,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+        )
+
+
+class BaseBackendUserPreference(_BaseUserPreferenceModel):
+    preference_type: PreferenceType = Field(PreferenceType.BACKEND, frozen=True)
+
+    class Config:
+        exclude_from_serialization: ClassVar[set[str]] = {
+            "preference_type",
+        }
+
+
+class BaseFrontendUserPreference(_BaseUserPreferenceModel):
+    preference_type: PreferenceType = Field(
+        default=PreferenceType.FRONTEND, frozen=True
+    )
+
+    # NOTE: below fields do not require storage in the DB
+    preference_identifier: PreferenceIdentifier = Field(
+        ..., description="used by the frontend"
+    )
+
+    class Config:
+        exclude_from_serialization: ClassVar[set[str]] = {
+            "preference_identifier",
+            "preference_type",
+        }
+
+
+class BaseUserServiceUserPreference(_BaseUserPreferenceModel):
+    preference_type: PreferenceType = Field(PreferenceType.USER_SERVICE, frozen=True)
+
+    # NOTE: preferences are stored per service and the version is not considered
+    service_key: ServiceKey = Field(
+        ..., description="the service which manages the preferences"
+    )
+
+    # NOTE: below fields do not require storage in the DB
+    last_changed_utc_timestamp: float = Field(
```

Comment:
why do you use a UNIX timestamp here? why not use a datetime which is more readable?